pkg/apis/eksctl.io/v1alpha4/vpc.go

```diff
@@ -92,22 +92,28 @@ func (c *ClusterConfig) PublicSubnetIDs() []string {
 
 // ImportSubnet loads a given subnet into cluster config
 func (c *ClusterConfig) ImportSubnet(topology SubnetTopology, az, subnetID, cidr string) error {
-	subnetCIDR, _ := ipnet.ParseCIDR(cidr)
-
 	if c.VPC.Subnets == nil {
 		c.VPC.Subnets = &ClusterSubnets{}
 	}
 
-	var subnets map[string]Network
 	switch topology {
 	case SubnetTopologyPrivate:
-		subnets = c.VPC.Subnets.Private
+		if c.VPC.Subnets.Private == nil {
+			c.VPC.Subnets.Private = make(map[string]Network)
```

Comment:
Just trying to understand what was going on here.

The cause of the issue seems to be https://github.com/weaveworks/eksctl/pull/513/files#diff-9cf0956b5581521f18ebe4dedc1f1552L90 - we had been mistakenly throwing away the imported `subnets` when `c.VPC.Subnets.Private` or `c.VPC.Subnets.Public`was empty?

Reply:
Thanks for asking! I've looked into this some more, and actually have a better idea of what is going on.

I was assuming that maps are always treated as references, which is indeed the case. However, if you assign a nil map to another nil map, nothing will actually happen. Namely, maps are treated as references (well, sort of), but only after they were initialised. Take a look here: https://play.golang.org/p/OTCUWDoH9HK. I didn't realise this until now, or at least I forgot this.

Reply:
Actually, it's even simpler then this, any nil reference can be assigned to another one, but nothing gets copied really: https://play.golang.org/p/KFefMZEabWQ.

Reply:
So it sort of boils down to `i1 = i2` being the same as `i1 = nil`, so the compiler cannot really tell these apart, hence it has no way of telling me that I'm probably doing something wrong... TIL :)